.github/workflows/build-docs.yml

```
@@ -0,0 +1,60 @@
+name: Build Docs
+
+on:
+  workflow_dispatch:
+  push:
+    branches: ["main", "docs**"]
+  pull_request:
+
+permissions:
+  contents: write
+
+concurrency:
+  group: build-site-${{ github.ref_name }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    if: github.event_name != 'pull_request' || !startsWith(github.head_ref, 'docs')
+    
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v4
+      
+    - name: Set up Node.js
+      uses: actions/setup-node@v4
+      with:
+        node-version: 22 # jsdoc2md 4.0.1 doesn't work with node 23
+        
+    - name: Install dependencies
+      run: npm install
+
+    - name: Install pandoc
+      uses: r-lib/actions/setup-pandoc@v2
+
+    - name: Setup R
+      uses: r-lib/actions/setup-r@v2
+      with:
+        use-public-rspm: true
+    
+    - name: Install R Dependencies
+      uses: r-lib/actions/setup-r-dependencies@v2
+      with:
+        cache-version: 2
+        extra-packages: |
+          local::.
+        needs: |
+          website
+      
+    - name: Render docs website
+      working-directory: docs
+      run: source build-site.sh
+
+    - name: Commit changes
+      run: |
+        git config --global user.name "$GITHUB_ACTOR"
+        git config --global user.email "$GITHUB_ACTOR@users.noreply.github.com"
+        git add docs/
+        git commit -m "ci(docs): Automated site build"
+        git push
```

I believe if the add fails, this will not try to commit and then push

```suggestion
        git add docs/ && git commit -m "ci(docs): Automated site build" && git push || echo "No docs/ changes to commit"
```

---

Ah yeah, good call, thanks!